app/routes/session.py

```diff
@@ -50,39 +55,58 @@ def login():
 
     validate_jti(decrypted_token)
 
-    try:
-        runner_claims = validate_runner_claims(decrypted_token)
-    except ValidationError as e:
-        raise InvalidTokenException("Invalid runner claims") from e
+    runner_claims = get_runner_claims(decrypted_token)
+
+    metadata = MetadataProxy.from_dict(runner_claims)
+
     # pylint: disable=assigning-non-slot
-    g.schema = load_schema_from_metadata(metadata=runner_claims)
+    g.schema = load_schema_from_metadata(metadata=metadata)
     schema_metadata = g.schema.json["metadata"]
 
-    try:
-        questionnaire_claims = validate_questionnaire_claims(
-            decrypted_token, schema_metadata
-        )
-    except ValidationError as e:
-        raise InvalidTokenException("Invalid questionnaire claims") from e
+    questionnaire_claims = get_questionnaire_claims(
+        decrypted_token=decrypted_token, schema_metadata=schema_metadata
+    )
+
+    theme = g.schema.json["theme"]
+    questionnaire_id = None
+
+    if decrypted_token.get("version") == "v2":
+        runner_claims["survey_metadata"]["data"] = questionnaire_claims
 
-    claims = {**runner_claims, **questionnaire_claims}
+        data = runner_claims.get("survey_metadata").get("data", {})
+        ru_ref = data.get("ru_ref")
+        questionnaire_id = data.get("questionnaire_id")
+
+        claims = runner_claims
```

Comment:
Perhaps this is incorrect but some logic is needed as otherwise when it does `claims = {**runner_claims, **questionnaire_claims}` you would get questionnaire_claims duplicated at the the top level of the dict, they already exist in runner_claims["survey_metadata"]["data"]

Reply:
No longer an issue with the changes.